File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
           
        for(int i = 0; i < queries.length; i++){
            int s = queries[i][0];
            int e = queries[i][1];
            int k = queries[i][2];
            int result = Integer.MAX_VALUE;
            boolean check = true;
            
            for(int j = s; j <= e; j++){
                int n = arr[j];
                if(n > k){
                    result = result > n ? n : result;
                    check = false;
                }
            }
            
            if(check) result = -1;
            
            answer[i] = result;
        }

        return answer;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    
    for s, e, k in queries:
        n = 1000001
        for i in range(s, e + 1):
            if arr[i] > k:
                n = min(arr[i], n)
            
        if n == 1000001:
            answer.append(-1)
        else:
            answer.append(n)
    
    return answer